src/common/error.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <stdio.h>
+#include <stdarg.h>
+#include <stdlib.h>
```

Comment:
Fyi, in C++ it's common to use the headers `cstdio`, `cstdlib`, etc. which define everything in the `std` namespace. In this way, we can keep our global namespace in an absolutely pristine and untouched condition 😄